src/hooks/form-date-picker/useFormDatePickerProps.tsx

```diff
@@ -64,72 +45,62 @@ export default function useFormDatePickerProps({
     [icon],
   )
 
-  const textField = React.useCallback(
-    (params: React.PropsWithChildren<TextFieldProps>) => (
-      <StyledTextField
-        {...params}
-        id={id}
-        variant="outlined"
-        margin="dense"
-        size="small"
-        label={undefined}
-        placeholder={placeholder}
-        autoComplete={autocompleteAttributes}
-        aria-describedby={ariaDescribedby}
-        inputProps={{
-          ...params.inputProps,
-          className: clsx(
-            params.inputProps?.className,
-            'input ob-input',
-            className,
-          ),
-        }}
-        fullWidth
-        onBlur={(e) => {
-          //need to run the blur function being passed from the picker
-          params?.onBlur?.(e)
-          onBlur()
-        }}
-        //we have our own error and helper text state
-        error={undefined}
-      />
-    ),
+  const formDatePickerProps = React.useMemo(
+    () => ({
+      slots: {
+        openPickerIcon,
+      },
+      slotProps: {
+        actionBar: {
+          actions: [
+            'clear',
+            'today',
+            'cancel',
+            'accept',
+          ] as PickersActionBarAction[],
+        },
+        popper: { disablePortal: true },
+        textField: {
+          id,
+          fullWidth: true,
+          label: undefined,
+          margin: 'dense' as const,
+          size: 'small' as const,
+          placeholder: placeholder,
+          'aria-describedby': ariaDescribedby,
+          onBlur,
+          //we have our own error and helper text state
+          error: false,
+          inputProps: {
+            className: clsx('input ob-input', className),
+          },
+        },
+      },
+      onClose: onBlur,
+      onChange: (newDate: Date | null) => {
+        if (!(newDate instanceof Date) || isNaN(newDate.valueOf())) {
+          onChange(undefined)
+        } else {
+          onChange(newDate)
+        }
+      },
+      maxDate: maxDateMemo,
+      minDate: minDateMemo,
+      value: valueMemo,
+    }),
     [
       ariaDescribedby,
-      autocompleteAttributes,
       className,
       id,
+      maxDateMemo,
+      minDateMemo,
       onBlur,
+      onChange,
```

Comment:
We are passing in new `onChange` function every render to `useFormDatePickerProps()` so no point in memo'ing this. I dont think we gain lot anyway.

Reply:
mmmm yeah I did see that and thought about going up another level.